components/schemes/Color/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,144 @@
+ */
+@interface MDCSemanticColorScheme : NSObject <MDCColorScheming>
+
+// Redeclare protocol properties as readwrite
+@property(nonnull, readwrite, nonatomic) UIColor *primaryColor;
+@property(nonnull, readwrite, nonatomic) UIColor *primaryColorVariant;
+@property(nonnull, readwrite, nonatomic) UIColor *secondaryColor;
+@property(nonnull, readwrite, nonatomic) UIColor *errorColor;
+@property(nonnull, readwrite, nonatomic) UIColor *surfaceColor;
+@property(nonnull, readwrite, nonatomic) UIColor *backgroundColor;
+@property(nonnull, readwrite, nonatomic) UIColor *onPrimaryColor;
+@property(nonnull, readwrite, nonatomic) UIColor *onSecondaryColor;
+@property(nonnull, readwrite, nonatomic) UIColor *onSurfaceColor;
+@property(nonnull, readwrite, nonatomic) UIColor *onBackgroundColor;
+
+/**
+ Convenience initializer that calls @c initWithMaterialDefaults.
+ */
+- (nonnull instancetype)init;
+
+/**
+ Initializes an instance of MDCSemanticColorScheme with the Material baseline color values.
+ */
+- (nonnull instancetype)initWithMaterialDefaults;
+
+/**
+ Creates a new color scheme with the provided color parameters.
+
+ @param primaryColor Displayed most frequently across your app.
+ @param primaryColorVariant A tonal variation of @c primaryColor.
+ @param secondaryColor Accents select parts of your UI.
+ @param errorColor The color used to indicate error status.
+ @param surfaceColor The color of surfaces such as cards, sheets, menus.
+ @param backgroundColor The underlying color of an app’s content.
+ @param onPrimaryColor A color that passes accessibility guidelines for text/iconography when drawn
+        on top of @c primaryColor.
+ @param onSecondaryColor A color that passes accessibility guidelines for text/iconography when
+        drawn on top of @c secondaryColor.
+ @param onSurfaceColor A color that passes accessibility guidelines for text/iconography when
+        drawn on top of @c surfaceColor.
+ @param onBackgroundColor A color that passes accessibility guidelines for text/iconography when
+        drawn on top of @c backgroundColor.
+
+ @returns a new color scheme with the provided values.
+ */
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor
+                         primaryColorVariant:(nonnull UIColor *)primaryColorVariant
+                              secondaryColor:(nonnull UIColor *)secondaryColor
+                                  errorColor:(nonnull UIColor *)errorColor
+                                surfaceColor:(nonnull UIColor *)surfaceColor
+                             backgroundColor:(nonnull UIColor *)backgroundColor
+                              onPrimaryColor:(nonnull UIColor *)onPrimaryColor
+                            onSecondaryColor:(nonnull UIColor *)onSecondaryColor
+                              onSurfaceColor:(nonnull UIColor *)onSurfaceColor
+                           onBackgroundColor:(nonnull UIColor *)onBackgroundColor
+    NS_DESIGNATED_INITIALIZER;
```

Comment:
Do we expect anyone to use this over init + properties?
What I would expect in the future is something like
- (instancetype)initWithJSONScheme: or the like.

If we keep it, please remember to add any add properties from above.